contracts/op-finality-gadget/src/state/finality.rs

```diff
@@ -1,8 +1,11 @@
-use cw_storage_plus::Map;
+use cw_storage_plus::{Item, Map};
 use std::collections::HashSet;
 
 /// Map of signatures by block height and fp
 pub(crate) const SIGNATURES: Map<(u64, &str), Vec<u8>> = Map::new("fp_sigs");
 
 /// Map of (block height, block hash) tuples to the list of fps that voted for this combination
 pub(crate) const BLOCK_VOTES: Map<(u64, &[u8]), HashSet<String>> = Map::new("block_hashes");
+
+/// Ordered list of forked blocks [(start_height_1, end_height_1), (start_height_2, end_height_2), ...]
+pub(crate) const FORKED_BLOCKS: Item<Vec<(u64, u64)>> = Item::new("forked_blocks");
```

Comment:
This structure means that the vector of start/end heights is considered as a single item. If we track a lot of forks (which might be the case), then the DB ops over it could be some bottleneck.

Would we consider other structures like `Map<u64, u64>` where key is the start height and value is the end height?

Reply:
> then the DB ops over it could be some bottleneck.

i think if it's ordered with no overlap, in practice it's gonna be efficient b/c we will loop in a reverse order. and it's unlikely the FG will need to read a lot of the old data to find the one its looking for

Reply:
If we store it as a map, i don't know how to design an efficient algorithm to decide if a block is reorged or not

Reply:
I think `prefix_range` or other iterator functions can be useful here. Check out `prefixed_range_works` under cw-storage-plus` repo

Reply:
i just took a look. i think the mental model will be cleaner to simply use (u64, u64) and query with a naive binary search like this

<img width="1140" alt="image" src="https://github.com/user-attachments/assets/d3a0bc64-a9ba-4ac6-a052-398d0152fe7a">

Reply:
Actually there is a much simpler way: given a height, use `prefix_range` to find the first key that is smaller than the height, and then check if the height is between the key and value